apps/react/src/features/studios/pages/StudioPage/StudioPage.tsx

```diff
@@ -0,0 +1,77 @@
+import { memo, FC, useEffect } from 'react';
+import IconButton from '@mui/material/IconButton';
+import { useSelector } from 'react-redux';
+import { selectIsDrawerOpen } from '@js-camp/react/store/drawer/selectors';
+import clsx from 'clsx';
+import { useAppDispatch, useAppSelector } from '@js-camp/react/store';
+import { Outlet, useParams, useSearchParams } from 'react-router-dom';
+import {
+	selectCursor, selectHasMoreData, selectIsPaginationEvent, selectOrdering, selectSearchValue, selectStudios,
+} from '@js-camp/react/store/studio/selectors';
+import { StudioQueryParameters } from '@js-camp/core/models/studio-query-parameters.model';
+import { fetchStudios } from '@js-camp/react/store/studio/dispatchers';
+import { Typography } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+
+import { composeQueryParams } from '../../utils/composeQueryParams';
+
+import { StudiosList } from '../../components/StudioList';
+
+import styles from './StudioPage.module.css';
+
+const StudioPageComponent: FC = () => {
+	const open = useSelector(selectIsDrawerOpen);
+	const { studioId } = useParams<{ studioId: string; }>();
+	const dispatch = useAppDispatch();
+	const studios = useAppSelector(selectStudios);
+	const cursor = useAppSelector(selectCursor) ?? undefined;
+	const hasMoreData = useAppSelector(selectHasMoreData);
+	const isPagination = useAppSelector(selectIsPaginationEvent);
+	const searchValue = useAppSelector(selectSearchValue);
+	const ordering = useAppSelector(selectOrdering);
+
+	const [searchParams, setSearchParams] = useSearchParams();
+
+	const params: StudioQueryParameters = Object.fromEntries(searchParams);
+
+	useEffect(() => {
+		setSearchParams(composeQueryParams({
+			...searchParams,
+			ordering: ordering ?? searchParams.get('ordering') ?? undefined,
+			search: searchValue ?? searchParams.get('search') ?? undefined,
+			cursor: cursor ?? searchParams.get('cursor') ?? undefined,
+		}));
+	}, [cursor, searchValue, ordering]);
+
+	useEffect(() => {
+		if (!isPagination || hasMoreData) {
+			dispatch(fetchStudios(params));
```

Comment:
I'm not sure whether your API request handlers configured correctly but anyway.
We should not forget about HTTP requests cancellation when user leaves the page.
In Angular we use `async` pipe which manages this stuff automatically. We take it for granted but React doesn't have such feature out of the box. We have to manage it manually.

In most cases it looks like this:
```suggestion
			const request = dispatch(fetchStudios(params));
			return request.abort;
```
It's only component related part. Our Redux store should be configured properly to make it work. For instance, we should provide `AbortSignal` from thunk to the API request.
You can read about it in [Canceling While Running](https://redux-toolkit.js.org/api/createAsyncThunk#canceling-while-running) section of the Redux Toolkit docs.